libminifi/src/core/yaml/YamlConfiguration.cpp

```diff
@@ -118,14 +119,18 @@ void YamlConfiguration::parseProcessorNodeYaml(YAML::Node processorsNode, core::
         }
         processor->setName(procCfg.name);
 
-        checkRequiredField(&procNode, "scheduling strategy",
-        CONFIG_YAML_PROCESSORS_KEY);
-        procCfg.schedulingStrategy = procNode["scheduling strategy"].as<std::string>();
+        auto strategyNode = getOptionalField(&procNode,
+                                             "scheduling strategy",
+                                             YAML::Node("EVENT_DRIVEN"),
```

Comment:
We should move this elsewhere.

Reply:
Move the Yaml::Node(...), or the getOptionalField call? And where to?

Reply:
Sorry I meant the YAML::Node. I shouldn't have been so terse. 

Here is a better way of putting it: Could we make it so that we centralized the defaults in one place so in case of change it's easier to track down? I know this isn't really the guise of the PR; however, it may improve the hygiene of this class to do so or make a follow on ticket.